src/server/frontend_wayland/wl_pointer.cpp

```diff
@@ -357,7 +356,12 @@ void WlSurfaceCursor::apply_latest_buffer()
             auto const cursor_image = std::make_shared<BufferCursorImage>(
                 *stream->lock_compositor_buffer(this),
                 hotspot);
-            surface->set_cursor_image(cursor_image);
+            if (cursor_image->as_argb_8888())
```

Comment:
If we're trying to do the thing only if the pixels were copied, perhaps it would be more obvious if we set a bool in the lambda, rather than relying on the logic here to be the same as the logic in `pixel_source->read()`. If there's some other reason for the if, document it.